src/prefect/tasks.py

```diff
@@ -777,7 +832,13 @@ def task(
     cache_key_fn: Callable[["TaskRunContext", Dict[str, Any]], Optional[str]] = None,
     cache_expiration: datetime.timedelta = None,
     retries: int = 0,
-    retry_delay_seconds: Union[float, int] = 0,
+    retry_delay_seconds: Union[
+        float,
+        int,
+        List[float],
+        Callable[["TaskRunContext", Dict[str, Any]], List[float]],
```

Comment:
I've made it this far, might as well keep suggesting
```suggestion
        Callable[[int], List[float]],
```